tour_manager/src/main/java/edu/artic/tours/manager/TourProgressManager.kt

```diff
@@ -23,4 +27,57 @@ class TourProgressManager {
     val selectedTour: Subject<Optional<ArticTour>> = BehaviorSubject.createDefault(Optional(null))
     val proposedTour: Subject<Optional<Pair<ArticTour, ArticTour.TourStop>>> = BehaviorSubject.createDefault(Optional(null))
     val leaveTourRequest: Subject<Boolean> = PublishSubject.create()
+
+    /**
+     * If [audioFileModel] belongs to currently selected tour i.e. [selectedTour],
+     * this method advances [selectedStop] to next tour stop.
+     */
+    @SuppressLint("CheckResult")
+    fun playBackEnded(audioFileModel: AudioFileModel) {
```

Comment:
Please add a `@WorkerThread` annotation to make it clear this shouldn't run on the main thread.

Reply:
Good idea. Updated.